executions/graphql-kotlin-transaction-batcher-instrumentation/src/main/kotlin/com/expediagroup/graphql/transactionbatcher/instrumentation/extensions/DataFetchingEnvironmentExtensions.kt

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright 2022 Expedia, Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.expediagroup.graphql.transactionbatcher.instrumentation.extensions
+
+import com.expediagroup.graphql.transactionbatcher.instrumentation.exceptions.MissingTransactionBatcherException
+import com.expediagroup.graphql.transactionbatcher.transaction.TransactionBatcher
+import graphql.schema.DataFetchingEnvironment
+
+/**
+ * get an instance of [TransactionBatcher] from the GraphQLContext
+ * @return a [TransactionBatcher] instance or [MissingTransactionBatcherException] if there
+ * is not a [TransactionBatcher] instance in the GraphQLContext
+ */
+fun DataFetchingEnvironment.transactionBatcher(): TransactionBatcher =
+    this.graphQlContext
+        .get<TransactionBatcher>(TransactionBatcher::class)
```

Comment:
do you actually need both the generic and the passed value? I thought our inline extension function should have simplified this?

Reply:
the extension function is defined in the generator library, in order to be able to use it i would need to import that library or define the extension function here, or maybe extract all these utility extensions in a dedicated artifact